### PR TITLE
Utku / Class 19 / Homework Week 3

### PR DESCRIPTION
**Pull request by u61**
All tests have been carried out.

### CODE REVIEW

File: Week3/homework/step3-bonus.js [hunk 1]
```diff
@@ -3,8 +3,8 @@
 const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
 
 function makeUnique(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const unique = arr.filter((x, y, z) => z.indexOf(x) === y);
+  return unique;
```
Comment (remarcmij) [inline Week3/homework/step3-bonus.js]: See my comment below with respect to creating the variable `unique`.

File: Week3/homework/step2-2.js [hunk 2]
```diff
@@ -2,19 +2,27 @@
 
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
-
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  let i = startIndex;
+  for (i; i <= stopIndex; i++) {
```
[the remark this answers is not in the document]
Comment (u61) [reply Week3/homework/step2-2.js]: Got it and put it inside of for loop as you show.

File: Week3/homework/step2-2.js [hunk 3]
```diff
@@ -2,19 +2,27 @@
...
+    if (i % 3 === 0 && i % 5 === 0) {
+      threeCallback(i);
+      fiveCallback(i);
+    } else if (i % 3 === 0) {
+      threeCallback(i);
+    } else if (i % 5 === 0) {
+      fiveCallback(i);
+    }
```
[the remark this answers is not in the document]
Comment (u61) [reply Week3/homework/step2-2.js]: I was also uncomfortable with these if statement block and then solved more logically as you say.

File: Week3/homework/step2-2.js [hunk 4]
```diff
@@ -2,19 +2,27 @@
...
+    numbers.push(i);
+  }
+  console.log('numbers:', numbers);
```
[the remark this answers is not in the document]
Comment (u61) [reply Week3/homework/step2-2.js]: You're right Sir. I changed sequence and added another for loop for creating array and iterating it.

File: Week3/homework/step2-2.js [hunk 5]
```diff
@@ -2,19 +2,27 @@
...
 }
 
 function sayThree(number) {
-  // Replace this comment and the next line with your code
-  console.log(number);
+  console.log(number, 'can divide by 3');
```
[the remark this answers is not in the document]
Comment (u61) [reply Week3/homework/step2-2.js]: Grammer error, corrected.

File: Week3/homework/step2-3.js [hunk 6]
```diff
@@ -5,8 +5,11 @@ function repeatStringNumTimesWithFor(str, num) {
   // eslint-disable-next-line
   let result = '';
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  if (num > 0) {
```
[the remark this answers is not in the document]
Comment (u61) [reply Week3/homework/step2-3.js]: Absolutely Sir. If statement deleted.

File: Week3/homework/step2-3.js [hunk 7]
```diff
@@ -31,8 +36,10 @@ function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line
   let result = '';
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  do {
```
[the remark this answers is not in the document]
Comment (u61) [reply Week3/homework/step2-3.js]: Got it. I couldn't think that. do-while loop implemented in if statement.

File: Week3/homework/step3.js [hunk 8]
```diff
@@ -1,8 +1,10 @@
 'use strict';
 
 function createBase(base) {
-  // Replace this comment and the next line with your code
-  console.log(base);
+  return function addSix(num) {
+    const sum = base + num;
+    return sum;
```
[the remark this answers is not in the document]
Comment (u61) [reply Week3/homework/step3.js]: I understand. It is shorthened.

File: Week3/homework/step3-bonus.js (hunk 1 above)
Comment (u61) [reply Week3/homework/step3-bonus.js]: Also this one is shorthened.

**Comment by u61** [comment]
Appreciated for your review Sir. Carefully read and corrected.

**Pull request closed without merging**